### DISCUSSION TITLE
Building for Win-ARM64

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#16030] [Q&A] [answered]
I want to build my Windows-Uno-app for arm64. But I get this error:

`error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.Mono.win-arm64. No packages exist with this id in source(s): C:\Program Files\dotnet\library-packs, Microsoft Visual Studio Offline Packages, NuGet official package source`

Does anybody know how to overcome this?

This is my build-command:

`dotnet publish mypp.Windows.csproj -c Release -f net8.0-windows10.0.19041.0 -p:Platform=arm64 -r win-arm64 --self-contained  /p:GenerateAppxPackageOnBuild=true`

**Comment by spouliot** [comment]
I think you're hitting dotnet issue https://github.com/dotnet/sdk/issues/21877

A quick workaround is to make sure you have only a single TFM (`net8.0-windows10.0.19041.0`) inside your `.csproj`.

An alternative is to use VSCode and the Uno extension which deals more cleanly (no need to edit the .csproj) with this issue.

**Reply by TopperDEL** [reply]
This is my csproj:

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net8.0-windows10.0.19041.0</TargetFramework>
    <TargetPlatformMinVersion>10.0.18362.0</TargetPlatformMinVersion>
    <RootNamespace>myApp</RootNamespace>
    <Platforms>x86;x64;arm64</Platforms>
```

So I already have only one TFM in there, right? And this also fails in my Github action build pipeline with the same error.

**Reply by spouliot** [reply]
hmm... it might be a different issue.

But if you have only one target framework then you should not need to specify it on the command-line. Did you try building without specifying `-f net8.0-windows10.0.19041.0` ?

**Reply by spouliot** [reply]
There's clearly a nuget bug since it's trying to retrieve `Microsoft.NETCore.App.Runtime.Mono.win-arm64` and there is **no** mono runtime for windows.

**Reply by TopperDEL** [reply]
Leaving `-f ...` out leads to the same error.

**Reply by TopperDEL** [reply]
If a binlog is helpful here: 
[msbuild.zip](https://github.com/unoplatform/uno/files/14759843/msbuild.zip)

**Reply by TopperDEL** [reply]
Hm, it seems not the Windows-csproj is the problem but the "shared" project. It had these TargetFrameworks:

```
<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows')) or '$(EnableWindowsTargeting)' == 'true'">$(TargetFrameworks);net8.0-windows10.0.19041</TargetFrameworks>
<TargetFrameworks>$(TargetFrameworks);net8.0;net8.0-maccatalyst</TargetFrameworks>
<TargetFrameworks Condition="'$(OverrideTargetFramework)'!=''">$(OverrideTargetFramework)</TargetFrameworks>
```

It seems the `net8.0-maccatalyst`-setting is raising the error.

**Reply by TopperDEL** [reply, marked as answer]
Yes, that seemed to be the reason. I changed my csproj for the shared-project to this:

```
<PropertyGroup>
    <IsWindows>$([MSBuild]::IsOSPlatform('Windows'))</IsWindows>
</PropertyGroup>
  <PropertyGroup>
		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows')) or '$(EnableWindowsTargeting)' == 'true'">$(TargetFrameworks);net8.0-windows10.0.19041</TargetFrameworks>
		<TargetFrameworks>$(TargetFrameworks);net8.0;</TargetFrameworks>
		<TargetFrameworks Condition="'$(OverrideTargetFramework)'!=''">$(OverrideTargetFramework)</TargetFrameworks>
		
	    <!-- Ensures the .xr.xml files are generated in a proper layout folder -->
		<GenerateLibraryLayout>true</GenerateLibraryLayout>
	</PropertyGroup>
	<PropertyGroup Condition="'$(IsWindows)' != 'true'">
    <TargetFrameworks>$(TargetFrameworks);net8.0-maccatalyst</TargetFrameworks>
</PropertyGroup>
```
